lib/scrabble.rb

```diff
@@ -1,4 +1,100 @@
 module Scrabble
-  # require your gems and classes here
-  # require_relative 'scrabble/whatevs'
+
+  # constant scoring for scrabble
+ LETTERS = {
+   1 => ["A", "E", "I", "O", "U", "L", "N", "R", "S", "T"],
+   2=> ["D", "G"],
+   3=> ["B", "C", "M", "P"],
+   4=> ["F", "H", "V", "W", "Y"],
+   5=> "K",
+   8=> ["J", "X"],
+   10=> ["Q", "Z"]
+ }
+
+  class Scrabble
+
+    def self.is_valid?(input_letters)
+    # confirming that input is valid, only accepting letters
+
+      if input_letters == []
+        return "please type something in."
+        exit
+      end
+
+      possible_letters = %w(Q W E R T Y U I O P A S D F G H J K L Z X C V B N M)
+
+      input_letters.each do |letter|
+
+        if possible_letters.include?(letter)
+          return input_letters
+        else
+          return "Error, not valid letters."
+          exit
+        end
+      end
+
+    end
+
+    def self.score(word)
+      total_points = 0
+      word.upcase!
+      # splitting word into an array of letters
+      input_letters = word.split(//)
+
+      # calling the validating method is_valid?(word)
+      is_valid?(input_letters)
+
+      # for each letter in the input_letters array, iterate over it
+      input_letters.each do |letter|
+        LETTERS.each do |points, letters|
+          # for each key/value in LETTERS constant, iterate over this
+          # if the input_letters letter matches the LETTERS letters in the value,
+          # then add the points key to total_points
+          if letters.include?(letter)
+            total_points += points
+          end
+        end
+      end
+
+      return total_points
+    end
+
+    def self.highest_score_from(array_of_words)
+    # return the word with the highest score (with caveats)
+      scrabble_hash = {}
+      hi_score_array = []
+
+      array_of_words.each do |word|
+        score = 0
+        # calling self.score to score the words from array
+        score = score(word)
```

Comment:
Using a variable with the same name as a method is confusing - in other languages it simply will not be permitted so try to avoid it.